From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Tue, 15 Aug 2023 22:10:31 +0200
Subject: tests/test-runner: Add `toggle_overview` command

This will emit the `overlay-key` key which will be handled by the test
shell.

Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3186
Applied-upstream: 44.4, commit:db924e85def8bb966d112b80c2ca5fd758916cab
---
 src/tests/test-runner.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/tests/test-runner.c b/src/tests/test-runner.c
index 282c4ec..ceccdbb 100644
--- a/src/tests/test-runner.c
+++ b/src/tests/test-runner.c
@@ -1301,6 +1301,15 @@ test_case_do (TestCase *test,
         BAD_COMMAND("Unknown preference %s", argv[1]);
       }
     }
+  else if (strcmp (argv[0], "toggle_overview") == 0)
+    {
+      MetaDisplay *display = meta_context_get_display (test->context);
+
+      if (argc != 1)
+        BAD_COMMAND ("usage: %s", argv[0]);
+
+      g_signal_emit_by_name (display, "overlay-key", 0);
+    }
   else
     {
       BAD_COMMAND("Unknown command %s", argv[0]);
